cabby/geo/walk_test.py

```diff
@@ -25,40 +25,38 @@
 
 class WalkTest(unittest.TestCase):
 
-  def setUp(self):
+    @classmethod
+    def setUpClass(self):
 
-    # Load map from disk.
-    self.map = map_structure.Map(
-      "Bologna", 18)
+        # Load map from disk.
+        self.map = map_structure.Map("Bologna", 18)
 
-  def testRouteCalculation(self):
-    start_point = Point(11.3414, 44.4951)
-    end_point = Point(11.3444, 44.4946)
-    route = walk.compute_route(start_point, end_point, self.map.
-    nx_graph, self.map.nodes)
+    def testRouteCalculation(self):
+        start_point = Point(11.3414, 44.4951)
+        end_point = Point(11.3444, 44.4946)
+        route = walk.compute_route(start_point, end_point, self.map.
+                                   nx_graph, self.map.nodes)
 
-    # Check the size of the route. 
-    self.assertEqual(route['geometry'].shape[0], 9)
+        # Check the size of the route.
+        self.assertEqual(route['geometry'].shape[0], 9)
 
-    # Check that the correct points are in the route.
-    first_point = util.tuple_from_point(route.iloc[0]['geometry'])
-    second_point = util.tuple_from_point(route.iloc[1]['geometry'])
-    self.assertEqual(first_point, (44.4950415, 11.341228))
-    self.assertEqual(second_point, (44.4948365, 11.3424075))
+        # Check that the correct points are in the route.
+        first_point = util.tuple_from_point(route.iloc[0]['geometry'])
+        second_point = util.tuple_from_point(route.iloc[1]['geometry'])
+        self.assertEqual(first_point, (44.4950415, 11.341228))
+        self.assertEqual(second_point, (44.4948365, 11.3424075))
 
+    def testPointsSelection(self):
```

Comment:
This test is a victim of non-determinism and seems likely to either miss errors in the future or be a source of hard-to-track failures. Can you change the test to make it so you can check for specific values? You may need to refactor things a bit, e.g. by expanding get_points_and_route into two functions.

Reply:
Still need to address this comment!

Reply:
OkK I added a random seed to solve this